src/tree_cmd.cpp

```diff
@@ -80,6 +82,30 @@ static bool CanPlantTreesOnTile(TileIndex tile, bool allow_desert)
 	}
 }
 
+/**
+* Tests if a tile is near an already existing forest.
+* Trees won't spread if not placed near some other trees.
+*
+* @param tile the tile of interest
+* @return true if the tile is near a forest
+*/
+static bool IsNearbyForest(TileIndex tile)
+{
+	uint planted_tile_count = 0;
+
+	/* An already planted tile can always be planted again. */
+	if (IsTileType(tile, MP_TREES)) return true;
+
+	/* Count the trees around the clear tile to determine if it's near a forest */
+	for (TileIndex t : TileArea(tile).Expand(FOREST_SEARCH_RADIUS)) {
+		if (IsTileType(t, MP_TREES)) {
+			planted_tile_count++;
+		}
+	}
+
+	return (planted_tile_count >= FOREST_THRESHOLD);
```

Comment:
The threshold is only 6, but you iterate for all 25 tiles.

You can early return true after incrementing `planted_tile_count` if it's reached the threshold, and then just return false at the end of the function.

This will avoid unnecessarily checking more tiles than necessary.